CHANGELOG.md

```diff
@@ -1,3 +1,108 @@
+ * `ChannelManager` supports sending and receiving short channel id aliases in
+   the `funding_locked` message. These are used when forwarding payments and
+   constructing invoice route hints for improved privacy (#1311).
+ * `ChannelDetails` now has a `inbound_scid_alias` field and
+   `get_inbound_payment_scid` method (#1311).
+ * `ChannelManager` no longer fails a channel when receiving redundant
+   `funding_locked` messages since they may be used to update the channel's
+   short channel id alias (#1311).
+ * `Features` mutator methods now take `self` by reference instead of by value
+   (#1331).
+ * Invoice creation functions in `lightning-invoice` crate's `utils` module
+   filter invoice hints in order to limit the invoice size (#1325).
+ * Invoice creation functions in `lightning-invoice` crate's `utils` module
+   include versions that accept a description hash instead of only a description
+   (#1361).
+ * `RoutingMessageHandler::sync_routing_table` has been renamed `peer_connected`
+   (#1368).
+ * `MessageSendEvent::SendGossipTimestampFilter` has been added to indicate that
+   a `gossip_timestamp_filter` should be sent (#1368).
+ * `PeerManager` takes an optional `NetAddress` in `new_outbound_connection` and
+   `new_inbound_connection`, used to report back the remote address to the
+   connecting peer in the `init` message (#1326).
+ * `ChannelManager::accept_inbound_channel` now takes a `user_channel_id`, which
+   is given back in `Event::ChannelClosed::user_channel_id` (#1381).
+ * `ProbabilisticScoringParameters` now has a `base_penalty_msat` option, which
+   default to 500 msats. It is applied at each hop to help avoid longer paths
+   (#1375).
+ * `ProbabilisticScoringParameters::liquidity_penalty_multiplier_msat`'s default
+   value is now 40,000 msats instead of 10,000 msats (#1375).
+ * The `lightning` crate has a `grind_signatures` feature used to produce
+   signatures with low r-values for more predictable transaction weight. This
+   feature is on by default (#1388).
+ * Lightning feature `option_scid_alias` is now supported and negotiated when
+   opening a channel with a peer. It can be configured via
+   `ChannelHandshakeConfig::negotiate_scid_privacy` and is off by default.
+   (#1351).
+ * `OpenChannelRequest` now has a `channel_type` field indicating the features
+   the channel will operate with.
+
+## Bug Fixes
+ * Fixed a compilation error in `ProbabilisticScorer` under `--feature=no-std`
+   resulting from `core` not having `f64::log10`. A log approximation is used
+   instead independent of feature settings (#1347).
+ * Fixed an integer underflow in `find_route` when
+   `RouteParameters::final_cltv_expiry_delta` exceeds
+   `RouteParameters::max_total_cltv_expiry_delta` (#1358).
+ * Fixed a bug where a `funding_locked` message was delayed by a block if the
+   funding transaction was confirmed while offline (#1363).
+ * `NetGraphMsgHandler` now send a `gossip_timestamp_filter` message upon
```

Comment:
Do users need to care what message is sent and why? Presumably they just care that we previously didn't continue receiving gossip updates after initial connection and now we do receive that data.